exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/VarLengthValuesColumn.java

```diff
@@ -81,6 +112,14 @@ public boolean skipReadyToReadPositionUpdate() {
     return false;
   }
 
+  // decimalLengths list is part of a near-term fix for DRILL-4184.
+  // Decimal[23]8SparseVector classes are fixed width vectors, without ability to "remember" offsets of
+  // (variable width) field sizes.  so, we "remember" the array sizes in decimalLengths (also used to
+  // "remember" whether a value was null, for nullable decimal columns).
+  // TODO: storage of decimal values should support variable length values in a much cleaner way than this,
+  // perhaps with a new variable width Decimal vector class.
+  protected ArrayList<Integer> decimalLengths = new ArrayList();
```

Comment:
I have moved decimalLengths class member to top of the class.  I found that the list is NOT always accessed at the end, hence left it as a list.